lib/chef/provider/launchd.rb

```diff
@@ -89,24 +89,41 @@ def gen_path_from_type
       end
 
       def manage_plist(action)
-        return unless manage_agent?(action)
```

Comment:
this resource terrifies me

Reply:
yeah it should be somewhat better after this